#1100: Study list test cases: Patient Name, MRN, Modality, Description

mirnasilva opened the pull request:
Test cases for issue #1100 
The majority of the tests are blocked by issues: #1114 #1120 

Hardcoded tests variations for case matching (camel, upper, lower and mixed), to guarantee that we will get always the same expected result.

mirnasilva commented on platform/viewer/cypress/integration/common/OHIFStudyList.spec.js:
```diff
@@ -0,0 +1,181 @@
+const expectExport = require('expect');
```
need to remove this

mirnasilva commented on platform/viewer/cypress/integration/common/OHIFStudyViewer.spec.js:
```diff
@@ -72,30 +72,26 @@ describe('OHIF Study Viewer Page', function() {
-  //   cy.get('.measurementLocation')
-  //     .should('contain.text', descriptionText);
-  // });
+  it('checks if Description can be added to measurement item under Measurements panel', () => {
```
issue #1013 was fixed. Uncommenting this test.

mirnasilva commented:
Pending tests for Sort Results and Pagination.

dannyrb commented:
@mirnasilva, studylist changes were merged yesterday. You may need to pull latest and verify your selectors are still accurate.

mirnasilva commented:
- Fixed failures on CI
- Added tablet tests
- Refactored aliases and desktop tests
- Added `cy.waitStudyList()` custom command to wait for the search results

dannyrb commented on platform/viewer/cypress/support/commands.js:
```diff
@@ -76,12 +78,18 @@ Cypress.Commands.add('isPageLoaded', (url = '/viewer/') => {
 
+Cypress.Commands.add('waitStudyList', () => {
+  cy.get('@searchResult').should($list => {
+    expect($list).to.not.have.class('no-hover');
```
I think checking the list length may be more reliable (if this class changes).

mirnasilva replied:
The problem with length is that if we don't find results or the result is loading, the length will be 1. This can be misled if our search result also returns 1.

dannyrb commented on platform/viewer/cypress/support/aliases.js:
```diff
@@ -52,3 +52,28 @@ export function initVTKToolsAliases() {
+
+//Creating aliases for Study List page elements on Tablet experience
+export function initStudyListAliasesOnTablet() {
+  cy.get('.study-count').as('studyCount');
```
You could use these methods to resize to the correct width before setting aliases?

mirnasilva replied:
Yes, the resizing is being done before calling this function.

dannyrb commented on platform/viewer/cypress/integration/common/OHIFStudyList.spec.js:
```diff
@@ -0,0 +1,328 @@
+      cy.viewport(1750, 720);
+      cy.openStudyList();
+      cy.initStudyListAliasesOnDesktop();
+      // run these tests as if in a desktop
```
I think you're okay to remove this comment :+1:

dannyrb commented on platform/viewer/cypress/integration/common/OHIFStudyList.spec.js:
```diff
@@ -0,0 +1,328 @@
+      // run these tests as if in a desktop
+    });
+
+    // //TO-TO: This test should be uncommented once issue #1114 is fixed: https://github.com/OHIF/Viewers/issues/1114
```
I've responded to your issue with some additional context here:

- https://github.com/OHIF/Viewers/issues/1114#issuecomment-549216117

We may want to discuss how best to test given the new information, and if we should create new tickets/stories around improving the application to clear up any misconceptions users may have with search + search capabilities.

mirnasilva replied:
Got it. In order to do a clean up here, I'll remove all the commented tests related to `fuzzymatching`. Once we decide to activate that, we can always go back and write one simple test to check it.

dannyrb commented on platform/viewer/cypress/integration/common/OHIFStudyList.spec.js:
```diff
@@ -0,0 +1,328 @@
+      cy.get('@searchResult').should($list => {
+        expect($list.length).to.be.eq(17);
+      });
+    });
```
We can probably cut back to a single modality test.

dannyrb commented on platform/viewer/cypress/integration/common/OHIFStudyList.spec.js:
```diff
@@ -0,0 +1,328 @@
+  });
+
+  context('Tablet resolution', function() {
+    beforeEach(function() {
```
If any of these tests are identical, but with different selectors, we may want to look into a better pattern. Not a blocker for this PR; just something to think about.

mirnasilva replied:
I believe that leaving it separated will make it easier to identify possible failures. :)
